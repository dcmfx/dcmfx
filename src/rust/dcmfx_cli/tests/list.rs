mod utils;

use assert_cmd::Command;
use insta::assert_snapshot;
use utils::{get_stderr, get_stdout, to_native_path};

#[test]
fn with_multiple_directories() {
  let mut cmd = Command::cargo_bin("dcmfx_cli").unwrap();

  let assert = cmd
    .arg("list")
    .arg("../../../test/assets/fo-dicom")
    .arg("../../../test/assets/pydicom")
    .assert()
    .success();

  let mut lines: Vec<_> =
    get_stdout(assert).split("\n").map(to_native_path).collect();
  lines.sort();

  #[cfg(windows)]
  assert_snapshot!("with_multiple_directories_windows", lines.join("\n"));

  #[cfg(not(windows))]
  assert_snapshot!("with_multiple_directories", lines.join("\n"));
}

#[test]
fn with_summary() {
  let mut cmd = Command::cargo_bin("dcmfx_cli").unwrap();

  let assert = cmd
    .arg("list")
    .arg("../../../test/assets/fo-dicom")
    .arg("--extension")
    .arg("dcm")
    .arg("--summarize")
    .assert()
    .success();

  assert_eq!(
    get_stderr(assert).trim(),
    "Found 32 DICOM files, size: 11.2 MiB"
  );
}

#[test]
fn with_json_lines_format() {
  let mut cmd = Command::cargo_bin("dcmfx_cli").unwrap();

  let assert = cmd
    .arg("list")
    .arg(to_native_path("../../../test/assets/fo-dicom"))
    .arg(to_native_path("../../../test/assets/pydicom"))
    .arg("--format")
    .arg("json-lines")
    .assert()
    .success();

  let stdout = get_stdout(assert);
  let mut lines: Vec<_> = stdout.split("\n").collect();
  lines.sort();

  #[cfg(windows)]
  assert_snapshot!("with_json_lines_format_windows", lines.join("\n"));

  #[cfg(not(windows))]
  assert_snapshot!("with_json_lines_format", lines.join("\n"));
}

#[test]
fn with_invalid_directory() {
  let mut cmd = Command::cargo_bin("dcmfx_cli").unwrap();

  let assert = cmd.arg("list").arg("missing-directory").assert().failure();

  #[cfg(windows)]
  assert_snapshot!("with_invalid_directory_windows", get_stderr(assert));

  #[cfg(not(windows))]
  assert_snapshot!("with_invalid_directory", get_stderr(assert));
}
