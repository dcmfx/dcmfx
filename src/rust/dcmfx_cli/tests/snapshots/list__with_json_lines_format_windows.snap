---
source: dcmfx_cli/tests/list.rs
expression: "lines.join(\"\\n\")"
snapshot_kind: text
---
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\CR-ModalitySequenceLUT.dcm","size":1261500}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\CT-MONO2-16-ankle.dcm","size":525436}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\CT1_J2KI.dcm","size":14156}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\DICOMDIR.dcm","size":13796}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\D_CLUNIE_CT1_RLE_FRAGS.dcm","size":254932}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\GH064.dcm","size":58050}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\GH1146.dcm","size":2124}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\GH1261.dcm","size":127134}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\GH1308.dcm","size":528304}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\GH177_D_CLUNIE_CT1_IVRLE_BigEndian_ELE_undefinded_length.dcm","size":530836}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\GH178.dcm","size":2832}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\GH184.dcm","size":20542}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\GH220.dcm","size":454}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\GH223.dcm","size":702}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\GH342.dcm","size":558222}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\GH355.dcm","size":525436}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\GH487.dcm","size":532}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\GH538-jpeg1.dcm","size":74990}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\GH538-jpeg14sv1.dcm","size":257386}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\GH610.dcm","size":21418}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\HasFragmentSequence.dcm","size":1702398}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\IM-0001-0001-0001.dcm","size":203304}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\Issue1097_FailToOpenDeflatedFileWithSQ.dcm","size":553}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\JPEGwithIcon.dcm","size":62560}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\OutOfBoundsOverlay.dcm","size":1021034}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\RI6XDosiGap.dcm","size":1576330}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\TestPattern_Palette.dcm","size":258228}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\TestPattern_Palette_16.dcm","size":258972}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\TestPattern_RGB.dcm","size":769114}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\genFile.dcm","size":262834}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\mr_brucker.dcm","size":266310}
{"path":"..\\..\\..\\test\\assets\\fo-dicom\\multiframe.dcm","size":563864}
{"path":"..\\..\\..\\test\\assets\\pydicom\\charset_files\\chrArab.dcm","size":1892}
{"path":"..\\..\\..\\test\\assets\\pydicom\\charset_files\\chrFren.dcm","size":1890}
{"path":"..\\..\\..\\test\\assets\\pydicom\\charset_files\\chrFrenMulti.dcm","size":1938}
{"path":"..\\..\\..\\test\\assets\\pydicom\\charset_files\\chrGerm.dcm","size":1894}
{"path":"..\\..\\..\\test\\assets\\pydicom\\charset_files\\chrGreek.dcm","size":1890}
{"path":"..\\..\\..\\test\\assets\\pydicom\\charset_files\\chrH31.dcm","size":1950}
{"path":"..\\..\\..\\test\\assets\\pydicom\\charset_files\\chrH32.dcm","size":1960}
{"path":"..\\..\\..\\test\\assets\\pydicom\\charset_files\\chrHbrw.dcm","size":1890}
{"path":"..\\..\\..\\test\\assets\\pydicom\\charset_files\\chrI2.dcm","size":1934}
{"path":"..\\..\\..\\test\\assets\\pydicom\\charset_files\\chrJapMulti.dcm","size":1918}
{"path":"..\\..\\..\\test\\assets\\pydicom\\charset_files\\chrJapMultiExplicitIR6.dcm","size":1940}
{"path":"..\\..\\..\\test\\assets\\pydicom\\charset_files\\chrKoreanMulti.dcm","size":1898}
{"path":"..\\..\\..\\test\\assets\\pydicom\\charset_files\\chrRuss.dcm","size":1890}
{"path":"..\\..\\..\\test\\assets\\pydicom\\charset_files\\chrSQEncoding.dcm","size":520}
{"path":"..\\..\\..\\test\\assets\\pydicom\\charset_files\\chrSQEncoding1.dcm","size":520}
{"path":"..\\..\\..\\test\\assets\\pydicom\\charset_files\\chrX1.dcm","size":1910}
{"path":"..\\..\\..\\test\\assets\\pydicom\\charset_files\\chrX2.dcm","size":1904}
{"path":"..\\..\\..\\test\\assets\\pydicom\\palettes\\fall.dcm","size":4078}
{"path":"..\\..\\..\\test\\assets\\pydicom\\palettes\\hotiron.dcm","size":4754}
{"path":"..\\..\\..\\test\\assets\\pydicom\\palettes\\hotmetalblue.dcm","size":4776}
{"path":"..\\..\\..\\test\\assets\\pydicom\\palettes\\pet.dcm","size":4732}
{"path":"..\\..\\..\\test\\assets\\pydicom\\palettes\\pet20step.dcm","size":4790}
{"path":"..\\..\\..\\test\\assets\\pydicom\\palettes\\spring.dcm","size":4104}
{"path":"..\\..\\..\\test\\assets\\pydicom\\palettes\\summer.dcm","size":4100}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\693_J2KI.dcm","size":3590}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\CT_small.dcm","size":39206}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\ExplVR_BigEnd.dcm","size":15412}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\GDCMJ2K_TextGBR.dcm","size":30706}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\J2K_pixelrep_mismatch.dcm","size":138518}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\JPEG-lossy.dcm","size":9844}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\JPEG2000-embedded-sequence-delimiter.dcm","size":3308}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\JPEG2000.dcm","size":3308}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\JPEGLSNearLossless_08.dcm","size":644}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\JPEGLSNearLossless_16.dcm","size":738}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\JPGExtended.dcm","size":9844}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\MR_small.dcm","size":9830}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\MR_small_RLE.dcm","size":7790}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\MR_small_bigendian.dcm","size":9708}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\MR_small_expb.dcm","size":9846}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\MR_small_implicit.dcm","size":9702}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\MR_small_jp2klossless.dcm","size":6008}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\MR_small_jpeg_ls_lossless.dcm","size":6124}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\MR_small_padded.dcm","size":9958}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_jpeg_no_color_transform.dcm","size":4316}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_jpeg_no_color_transform_2.dcm","size":4310}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_dcmtk_+eb+cr.dcm","size":3626}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_dcmtk_+eb+cy+n1.dcm","size":3136}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_dcmtk_+eb+cy+n2.dcm","size":3090}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_dcmtk_+eb+cy+np.dcm","size":3140}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_dcmtk_+eb+cy+s2.dcm","size":3094}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_dcmtk_+eb+cy+s4.dcm","size":3420}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_gdcm_KY.dcm","size":2998}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_jls_lossy_line.dcm","size":1100}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_jls_lossy_sample.dcm","size":1122}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_jpeg_app14_dcmd.dcm","size":4310}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_jpeg_dcmd.dcm","size":197506}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_jpeg_dcmtk.dcm","size":3424}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_jpeg_gdcm.dcm","size":5204}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_jpeg_lossy_gdcm.dcm","size":4982}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_rle.dcm","size":2006}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_rle_16bit.dcm","size":2606}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_rle_16bit_2frame.dcm","size":3896}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_rle_2frame.dcm","size":2696}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_rle_32bit.dcm","size":3760}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_rle_32bit_2frame.dcm","size":6246}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_small_odd.dcm","size":1444}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_small_odd_big_endian.dcm","size":1444}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_rgb_small_odd_jpeg.dcm","size":2044}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\SC_ybr_full_422_uncompressed.dcm","size":21686}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\UN_sequence.dcm","size":674}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\77654033\\CR1\\6154.dcm","size":2300}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\77654033\\CR2\\6247.dcm","size":2298}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\77654033\\CR3\\6278.dcm","size":2298}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\77654033\\CT2\\17106.dcm","size":3810}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\77654033\\CT2\\17136.dcm","size":3812}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\77654033\\CT2\\17166.dcm","size":3812}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\77654033\\CT2\\17196.dcm","size":3812}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892001\\CT2N\\6293.dcm","size":3920}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892001\\CT2N\\6924.dcm","size":3908}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892001\\CT5N\\2062.dcm","size":3936}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892001\\CT5N\\2392.dcm","size":3936}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892001\\CT5N\\2693.dcm","size":3936}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892001\\CT5N\\3023.dcm","size":3936}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892001\\CT5N\\3353.dcm","size":3938}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892003\\MR1\\15820.dcm","size":2336}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892003\\MR1\\4919.dcm","size":2336}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892003\\MR1\\5641.dcm","size":2330}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892003\\MR2\\15970.dcm","size":2336}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892003\\MR2\\4950.dcm","size":2356}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892003\\MR2\\4981.dcm","size":2354}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892003\\MR2\\5011.dcm","size":2354}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892003\\MR2\\6273.dcm","size":2348}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892003\\MR2\\6605.dcm","size":2348}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892003\\MR2\\6935.dcm","size":2350}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892003\\MR700\\4467.dcm","size":2350}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892003\\MR700\\4528.dcm","size":2348}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892003\\MR700\\4558.dcm","size":2348}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892003\\MR700\\4588.dcm","size":2350}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892003\\MR700\\4618.dcm","size":2350}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892003\\MR700\\4648.dcm","size":2350}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\98892003\\MR700\\4678.dcm","size":2350}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\DICOMDIR-bigEnd.dcm","size":11116}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\DICOMDIR-empty.dcm","size":396}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\DICOMDIR-implicit.dcm","size":11110}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\DICOMDIR-nooffset.dcm","size":11092}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\DICOMDIR-nopatient.dcm","size":11116}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\DICOMDIR-reordered.dcm","size":11116}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\DICOMDIR.dcm","size":11116}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\DICOMDIR.dcm","size":13066}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM000000.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM000001.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM000002.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM000003.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM000004.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM000005.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM000006.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM000007.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM000008.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM000009.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000A.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000B.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000C.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000D.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000E.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000F.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000G.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000H.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000I.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000J.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000K.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000L.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000M.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000N.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000O.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000P.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000Q.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000R.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000S.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000T.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000U.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000V.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000W.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000X.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000Y.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00000Z.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM000010.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM000011.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM000012.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM000013.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM000014.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM000015.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM000016.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM000017.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM000018.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM000019.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00001A.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00001B.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00001C.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\dicomdirtests\\TINY_ALPHA\\PT000000\\ST000000\\SE000000\\IM00001D.dcm","size":740}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\empty_charset_LEI.dcm","size":276}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\examples_jpeg2k.dcm","size":153760}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\examples_overlay.dcm","size":321700}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\examples_palette.dcm","size":283486}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\examples_rgb_color.dcm","size":231710}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\examples_ybr_color.dcm","size":224902}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\image_dfl.dcm","size":4637}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\liver_1frame.dcm","size":37084}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\liver_expb_1frame.dcm","size":36532}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\liver_nonbyte_aligned.dcm","size":101878}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\meta_missing_tsyntax.dcm","size":317}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\nested_priv_SQ.dcm","size":343}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\no_meta_group_length.dcm","size":408}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\priv_SQ.dcm","size":546}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\reportsi.dcm","size":2968}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\reportsi_with_empty_number_tags.dcm","size":2700}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\rtdose.dcm","size":7568}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\rtdose_1frame.dcm","size":1958}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\rtdose_expb.dcm","size":7618}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\rtdose_expb_1frame.dcm","size":2008}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\rtdose_rle.dcm","size":6816}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\rtdose_rle_1frame.dcm","size":2122}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\rtplan.dcm","size":2672}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\test-SR.dcm","size":6796}
{"path":"..\\..\\..\\test\\assets\\pydicom\\test_files\\waveform_ecg.dcm","size":291088}
